src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -513,6 +528,78 @@ public HashSet<Guid> Dependencies
             }
         }
 
+        /// <summary>
+        /// Gathers the packages that this graph depends on
+        /// </summary>
+        public IEnumerable<IPackage> PackageDependencies
+        {
+            get
+            {
+                // Get all packages installed locally
+                var localPackages = GetLocalPackages();
+                // Get set of assemblies used by nodes in this workspace
+                var assembliesUsed = GetAssembliesUsed();
+
+                // Create a dictionary that maps assembly names to the package they are contained in
+                var assemblyPackageDict = new Dictionary<string, IPackage>();
```

Comment:
you may want to actually use assembly objects here - including assembly versions, I have a feeling that will be useful...

Reply:
or AssemblyName objects.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname?view=netframework-4.8

Reply:
One thing that seems tricky to me here is that we currently only have the assembly _path_ from `AssembliesUsed` (which I'm getting from Library Services). So to look up packages in the dictionary we would first need a way to get the assembly `Version` so we can make the `AssemblyName` object.

Reply:
We could use this? 

[https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.loadfrom?view=netframework-4.8](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.loadfrom?view=netframework-4.8)

Would that hurt performance?

Reply:
the assemblies should already be loaded ... and if not we don't want to load them - we might want to load them into a reflection only context - also yes, loading assemblies that are not already loaded will be slow.

Reply:
Instead, I think what we want to do is see if the assembly is already loaded into the current app domain.

https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.getassemblies?view=netframework-4.8#System_AppDomain_GetAssemblies

this can also be slow - but faster than re loading from disk.

I think this is what we want though ⭐️ 
https://stackoverflow.com/questions/30306702/assemblyinfo-without-loading

Reply:
Nevermind.
https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname.getassemblyname?view=netframework-4.8

Reply:
that may not work for all assemblies - only those with assembly manifests - in the case of packages - that are being used in the graph, we may be able to make the assumption that the dll is already loaded into the appdomain...

Reply:
I agree with that assumption. Moving forward with this strategy.